### PR TITLE
dialog: adding maxWidth to `SingleTextInputDialog`

### PR DESCRIPTION
**Pull request by vladarama**
#### What it does

This PR adds a maxWidth attribute to `SingleTextInputDialogs`  in order to fix a styling issue that would occur when the input validation error message would get too long. This would cause the dialog to extend indefinitely. This fix adds a max-width of 400 px to the dialogs to make their styling more user-friendly. The modified dialogs include: `New File`, `New Folder`, `Rename` and `EditKeybinding` dialogs.

Before:
no maxwidth

After:
maxwidth

#### How to test

1. Open Theia application
2. Try causing errors while renaming files, creating new files/folders, modifying keybindings
3. The dialog should have a `max-width: 400px` css style
4. Long error messages should not extend the dialog indefinitely

Easy errors to test include: creating a file/folder with no name or a duplicate name, 
renaming a file to the same name, creating empty key bindings ...

Should be merged after #12585 

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/core/src/browser/dialogs.ts [hunk 1]
```diff
@@ -166,8 +166,10 @@ export abstract class AbstractDialog<T> extends BaseWidget {
         container.classList.add('dialogBlock');
         if (props.maxWidth === undefined) {
             container.setAttribute('style', 'max-width: none');
-        } else {
+        } else if (props.maxWidth < 400) {
             container.setAttribute('style', `max-width: ${props.maxWidth}px; min-width: 0px`);
+        } else {
+            container.setAttribute('style', `max-width: ${props.maxWidth}px`);
```
Comment (msujew) [inline packages/core/src/browser/dialogs.ts:172]: Question: Why would we want to remove the `min-width` style in case `max-width` is `>= 400`?

Comment (vladarama) [reply packages/core/src/browser/dialogs.ts:172]: The dialogs already have a minimum-width of 400 px from here:
From my understanding, the only case where we would need to override this `min-width: 400px` and set it to `0px` would be when we have a `max-width < 400`. In the other cases, we should keep the value from `.dialogBlock` and avoid overriding it which is why I removed the `min-width` style. Please tell me if I am wrong.
https://github.com/eclipse-theia/theia/blob/803efc994ef71352d36d4dfdc6fb87183cb9b6b7/packages/core/src/browser/style/dialog.css#L32-L40

Comment (msujew) [reply packages/core/src/browser/dialogs.ts:172]: I see, I didn't know about that. Makes sense now 👍

**Review by msujew — APPROVED** [review]
Alright, looks good to me 👍

**Pull request merged**